sklearn/cluster/k_means_.py

```diff
@@ -188,6 +189,12 @@ def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
         If a callable is passed, it should take arguments X, k and
         and a random state and return an initialization.
 
+    algorithm : "auto", "full" or "triangle_inequality", default="auto"
+        K-means algorithm to use. The classical EM-style algorithm is "full".
+        The "triangle_inequality" variation is more efficient by using the triangle
```

Comment:
I just checked, and unfortunately they are not always. I will double check what is happening. if they converge, they converge to the same solution, but if I stop after one iteration, they are different on iris. I'm not sure if this is a different choice of argmax or what happens there.